pkg/cloudprovider/providers/aws/device_allocator.go

```diff
@@ -40,11 +45,27 @@ type DeviceAllocator interface {
 	// name. Only the device suffix is returned, e.g. "ba" for "/dev/xvdba".
 	// It's up to the called to add appropriate "/dev/sd" or "/dev/xvd" prefix.
 	GetNext(existingDevices ExistingDevices) (mountDevice, error)
+
+	// LockForUse will lock the device allocator, so as another goroutine can't use it
+	LockForUse()
+
+	// Mark for use makes last selected device as taken
+	MarkForUse()
+
+	// Unlock the device allocator
+	Unlock()
+
+	// return current index and position
+	GetIndexPos() (int, int)
```

Comment:
This really feels like exposing our internals... if this is only for testing, use a cast to deviceAllocator instead...